tools/render-test/slang-support.cpp

```diff
@@ -105,16 +105,16 @@ RefPtr<ShaderProgram> ShaderCompiler::compileProgram(
     for (auto typeName : request.entryPointGenericTypeArguments)
         rawEntryPointTypeNames.Add(typeName.Buffer());
 
-    UInt globalExistentialTypeCount = request.globalExistentialTypeArguments.Count();
-    for( UInt ii = 0; ii < globalExistentialTypeCount; ++ii )
+    const int globalExistentialTypeCount = int(request.globalExistentialTypeArguments.Count());
```

Comment:
Nitpick 1: I get why you are using `int` here instead, but we should probably do a global refactor sooner or later to change most of the `UInt` stuff in APIs like `List<T>` and `String` over to `Int`, and then make a global policy decision about what we want our "natural integer type" to be, and make `Int` a `typedef` of that.

While `int` is nominally the preferred machine integer type in C/C++ in practice it seems like newer languages (Rust, Swift) are going in the direction of always making the "default" integer type match the pointer size of the machine, and that seems to be motivated by feedback from compiler writers about what lets them optimize best. The fact that `int` is 32-bit on 64-bit Windows seems to be more of a backwards-compatibility thing and less of a "this is the right default" thing.

Nitpick 2: whenever possible please refrain from including cleanups to code unrelated to the purpose of a PR. Since I'm deep in work on stuff related to existential type parameters/arguments, it is not unreasonable to imagine that these changes will cause merge conflicts for me when I try to update my branch from `master`.

Reply:
(To be clear, I concede the point that expert opinion is strongly in favor of signed integers being the default currency type for APIs without a really strong case for needing unsigned integers, and the Slang `core` libraries are in violation of those expert opinions.)

Reply:
Update: I guess I'm wrong about where Rust landed. They used to have machine-pointer-size `int` and `uint` types, but their current documentation implies they got renamed to `isize` and `usize` and they explicitly give the guideline that when in doubt programmers should default to `i32` (with an argument that it will usually the same speed or faster than `isize`)

Swift definitely has `Int` and `UInt` types that are machine-pointer-sized, which frees them from the burden of having to cast between machine-pointer-size and default-currency-type integers in most cases.

It guess there isn't enough of a consensus for me to call this a settled issue, but my gut instinct would be to define `Int` as pointer-sized (which it is) and make most internal interfaces just use `Int` across the board.